### PR TITLE
Added bookings controller

### PR DESCRIPTION
**Pull request by worksofgenius**
Please check this controller. I'm not confident in it.

### CODE REVIEW

File: app/controllers/bookings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    @listing = Listing.find(params[:listing_id])
+    @booking = Booking.new
+  end
+
+  def create
+    @booking = Booking.new(booking_params)
+    @listing = Listing.find(params[:listing_id])
+    @booking.listing = @listing
+    if @booking.save
+      redirect_to booking_path(@booking), notice: 'Booking was successfully created!'
```
Comment (zuccamia) [inline app/controllers/bookings_controller.rb]: The route name will be `listing_booking_path`

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def edit
+    @booking = Booking.find(params[:id])
+  end
+
+  def update
+    @booking = Booking.find(params[:id])
+    if @booking.update(booking_params)
+      redirect_to booking_path(@booking), notice: 'Booking was successfully updated.'
```
Comment (zuccamia) [inline app/controllers/bookings_controller.rb]: Similar to above

File: app/controllers/bookings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @booking = Booking.find(params[:id])
+    @booking.destroy
+    redirect_to bookings_path, notice: 'Booking was successfully deleted.'
```
Comment (zuccamia) [inline app/controllers/bookings_controller.rb]: The path name here will be `listing_bookings_path`. Sorry we only had the routes.rb after you created this!

**Review by zuccamia — COMMENTED** [review]
Apart from the `path` name and `booking_params`, all looks good!

File: app/controllers/bookings_controller.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  private
+
+  def booking_params
+    params.require(:booking).permit(:user, :start_date, :end_date)
```
Comment (zuccamia) [inline app/controllers/bookings_controller.rb]: I think we have another param, `:notes`. I'm not entirely sure about :user; I will need to test with routes to see...

File: app/controllers/bookings_controller.rb (hunk 1 above)
Comment (worksofgenius) [reply app/controllers/bookings_controller.rb]: I made the requested change.

File: app/controllers/bookings_controller.rb (hunk 2 above)
Comment (worksofgenius) [reply app/controllers/bookings_controller.rb]: I made the requested change.

File: app/controllers/bookings_controller.rb (hunk 3 above)
Comment (worksofgenius) [reply app/controllers/bookings_controller.rb]: No worries! Thanks for looking it over. I made the requested change.

File: app/controllers/bookings_controller.rb (hunk 4 above)
Comment (worksofgenius) [reply app/controllers/bookings_controller.rb]: I added `:notes`. I'm still not sure about `:user`...

**Pull request merged**